docs/src/index.md

```diff
@@ -525,11 +525,111 @@ To reference columns with more complicated expressions, you must wrap column ref
+
+```
+julia> @rtransform df :y = sum(AsTable(vars))
+2×4 DataFrame
+ Row │ a      b      c      y     
+     │ Int64  Int64  Int64  Int64 
+─────┼────────────────────────────
+   1 │    11     17     12     28
+   2 │    14     10      5     24
+```
+
+To subset all rows where the sum is greater than `25`, write
+
+```
+julia> @rsubset df sum(AsTable(vars)) > 25
+1×3 DataFrame
+ Row │ a      b      c     
+     │ Int64  Int64  Int64 
+─────┼─────────────────────
+   1 │    11     17     12
+```
+
+To understand the how this works, recall that DataFrames.jl allows for
+`AsTable(cols)` to be a `source` in a `source => fun => dest` mini-language
+expression. As a consequence, the transformation call
+
+```
+:y = f(AsTable(cols)) 
+```
+
+becomes
+
+```
+AsTable(cols) => f => :y
+```
+
+Note that DataFrames does *not* allow `source => fun => dest` commands 
+to be of the form 
+
+```
+[AsTable(cols), :x] => f => :y
+```
+
+As a consequence, DataFramesMeta.jl does not allow any other column selectors to appear 
+inside the expression. The command
+
+```
+:y = sum(AsTable(cols)) + :d
+```
+
+will fail. 
+
+Finally, note that everyting inside `AsTable` is escaped by default. There is no ned to use `$` inside `AsTable` on the right-hand-side. For example
+
+```
+:y = first(AsTable("a"))
+```
+
+will work as expected.  
+
+
+### AsTable and `@astable`, explained
+
+At this point we have seen `AsTable` appear in three places:
+
+1. `AsTable` on the left-hand-side of transformations: `$AsTable = f(:a, :b)`
+2. The macro-flag `@astable` within the transformation. 
+3. `AsTable(cols)` on the right hand side for multi-column transformations. 
+
+The differences between the three is summarized below
+
+| Operation         | Purpose                                                                             | Example           | Notes |
+|-------------------|-------------------------------------------------------------------------------------|-------------------|-------|
+| `$AsTable` on LHS | Create multiple columns at once, whose column names are only known programmatically | <pre lang=julia>$AsTable = f(:y)</pre> | Requires escaping with `$` until deprecation period ends for unquoted column names on LHS.      |
+| `@astable`        | Create multiple columns at once where number of columns is known in advance         | <pre lang="julia">@astable begin<br>    :y = :x - 1<br>    :z =  :y - 1<br>end</pre>
+| `AsTable` on RHS  | Work with multiple columns at once                                                  |<pre lang="julia">:y = sum(AsTable(Between("a", "z")))</pre> | Requires input columns, unlike on LHS |
+
+
+### Using `src => fun => dest` calls using `$`
+
+If an argument is entirely wrapped in `$()`, the result bypasses the anonymous function 
+creation of DataFramesMeta.jl and is passed to the underling DataFrames.jl function 
+directly. Importantly, this allows for `src => fun => dest` calls from the DataFrames.jl 
+"mini-language" directly.
```

Comment:
again - maybe add an example? something like `[:a, :b] .=> [sum mean]`? as I guess this is hard to express in DataFramesMeta.jl?